src/commands/git_secret_changes.sh

```diff
@@ -60,10 +65,28 @@ function changes {
     decrypted="${decrypted_x%x*}"
     # we ignore the exit code because _decrypt will _abort if appropriate.
 
+    if [[ 1 -eq $git ]]; then
+      _check_if_busybox_diff
+      # We output a merged text where the diffs are surrounded by git-style conflicts
+      diff_x=$(
+        diff -D AAAAAAA "${path}" <(echo -n "$decrypted") | \
```

Comment:
Ditto my comment above about simplicity/clarity

Reply:
This is the part which generates the conflicts blocks (git style) in case files are different (it outputs the full file if there are no changes).
According to the man page, `diff -D NAME  --ifdef=NAME  Output merged file to show `#ifdef NAME' diffs.`, that is diff will mark the begin and end of each change with C style ifdef / else / ifndef pre-compiler style blocks. AAAAAAA is the text/variable you want to use as marker. This is pretty nice, but as we are using git secret with, well, git, it would be nicer to have a git style markers.
For this reason, I pipe the output of the diff -D to multiple sed commands to remove the C style ifdef etc and substitute them with git style >>>>>>> ====== <<<<<< markers.
Normally, git would compare versions of a file, so you would have a commit hash or branch name, but we don't have this info, and the file name is the same, so I thought it was clearer to have the conflicts clearly state what's on disk and what's from the secret.

Reply:
I would really prefer if we didn't fetch one diff format and then convert it to another.